### ISSUE TITLE
Man pages do not specify that sssd dyndns_refresh_interval < 60 is pulled to 60 seconds

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#2205]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/2201
- Created at 2014-01-17 15:46:00 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed as Fixed
- Assigned to [amitkumar25nov](https://pagure.io/user/amitkumar25nov)
---

Ticket was cloned from Red Hat Bugzilla (product *Red Hat Enterprise Linux 7*): [Bug 1054777](https://bugzilla.redhat.com/show_bug.cgi?id=1054777)

    Description of problem:
    If sssd AD or IPA provider has dyndns_refresh_interval specified as less than
    60 seconds, actual refresh interval will be 60 seconds regardless.
    
    This is because of the update rate limit hard-coded in ad_dyndns_update_send
    and ipa_dyndns_update_send.
    
    Version-Release number of selected component (if applicable):
    python-sssdconfig-1.11.2-24.el7.noarch
    sssd-krb5-1.11.2-24.el7.x86_64
    sssd-krb5-common-1.11.2-24.el7.x86_64
    sssd-tools-1.11.2-24.el7.x86_64
    sssd-common-1.11.2-24.el7.x86_64
    sssd-ad-1.11.2-24.el7.x86_64
    sssd-1.11.2-24.el7.x86_64
    sssd-client-1.11.2-24.el7.x86_64
    sssd-ipa-1.11.2-24.el7.x86_64
    sssd-proxy-1.11.2-24.el7.x86_64
    sssd-common-pac-1.11.2-24.el7.x86_64
    sssd-ldap-1.11.2-24.el7.x86_64
    
    How reproducible:
    Always
    
    Steps to Reproduce:
    1. Assign an address to a dummy interface.
    2. Configure sssd for dynamic DNS updates either with AD or IPA provider.
    3. Specify the dummy interface with dyndns_iface.
    4. Specify 10 second refresh interval with dyndns_refresh_interval.
    5. Start sssd.
    6. Wait for initial dynamic DNS update.
    7. Change the dummy interface address.
    8. Wait 10 seconds.
    9. Check if the DNS records correspond to the interface address.
    10. Wait 50 seconds.
    11. Check if the DNS records correspond to the interface address.
    
    Actual results:
    First check: DNS records are outdated.
    Second check: DNS records are up-to-date.
    
    Expected results:
    First check: DNS records are up-to-date.
    Second check: DNS records are up-to-date.
    
    Additional info:
    It would suffice to specify this limit in the documentation, but a better fix
    would be making refresh interval less than 60 seconds effective, while
    preserving the 60 seconds limit for longer intervals.

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-01-23 14:57:49](https://pagure.io/SSSD/sssd/issue/2201#230544)**

Fields changed

blockedby:  => 
blocking:  => 
changelog:  => 
component: SSSD => Documentation
coverity:  => 
design:  => 
design_review:  => 0
feature_milestone:  => 
fedora_test_page:  => 
milestone: NEEDS_TRIAGE => SSSD 1.13 beta
review: True => 0
selected:  => 
testsupdated:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-26 18:49:58](https://pagure.io/SSSD/sssd/issue/2201#230545)**

Fields changed

mark:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-01-06 18:05:31](https://pagure.io/SSSD/sssd/issue/2201#230546)**

Fields changed

owner: somebody => preichl

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-10 14:15:53](https://pagure.io/SSSD/sssd/issue/2201#230547)**

Fields changed

milestone: SSSD 1.13 beta => SSSD 1.13 backlog
priority: major => trivial

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-12 20:36:29](https://pagure.io/SSSD/sssd/issue/2201#230548)**

Mass-moving tickets not planned for any immediate release and re-setting priority.

milestone: SSSD 1.13 backlog => SSSD Deferred
priority: trivial => major

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-10 22:51:16](https://pagure.io/SSSD/sssd/issue/2201#230549)**

Fields changed

keywords:  => easyfix
owner: preichl => somebody
priority: major => trivial
sensitive:  => 0
summary: Sssd dyndns_refresh_interval < 60 is pulled to 60 seconds => Man pages do not specify that sssd dyndns_refresh_interval < 60 is pulled to 60 seconds

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-15 11:54:10](https://pagure.io/SSSD/sssd/issue/2201#230550)**

I want to pick this up.

I hope these are steps which I need to follow
Owning the Ticket:
    Modify Ticket ==>    Blocked By: <my name>    ==> Submit Changes
   
Making Code Changes, Submitting
I hope these are steps to change, test, submit:
 1. Have latest sssd source code
     # vim version.m4
        # Primary version number
        m4_define([VERSION_NUMBER], [1.14.90])
 2. Reproduce the Issue. With steps mentioned in ticket.
 3. Apply the Fix, test whether issue is fixed (locally). Also make intgchk
 4. Submitting the patch:
        # git add <path-tofile>
        # git commit

Bugzilla and Fedora:
    Since RHEL is fork of fedora, so do we need to own Bugzilla ticket also while owning the Fedora/sssd ticket.
    Or We only change fedora/sssd ticket, code and it waterfalls to RHEL on upcoming release(if priotized)

Thanks

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-16 10:14:31](https://pagure.io/SSSD/sssd/issue/2201#230551)**

Replying to [comment:7 amitkumar25nov]:
> I want to pick this up.
> 
> I hope these are steps which I need to follow
> Owning the Ticket:
>     Modify Ticket ==>    Blocked By: <my name>    ==> Submit Changes
>    

Blocked by is not the right field, you should see "Action" below "Change properties" and in action you should see "accept". If not, I need to grant you more permissions, but for now I'll manually assign the ticket for you.

> Making Code Changes, Submitting
> I hope these are steps to change, test, submit:
>  1. Have latest sssd source code
>      # vim version.m4
>         # Primary version number
>         m4_define([VERSION_NUMBER], [1.14.90])
>  2. Reproduce the Issue. With steps mentioned in ticket.
>  3. Apply the Fix, test whether issue is fixed (locally). Also make intgchk

In general yes, but since you will only be changing a man page, I think it's sufficient to run make and then check with "man ./path/to/the/buildroot/src/man/file" that the man page looks OK.

>  4. Submitting the patch:
>         # git add <path-tofile>
>         # git commit
> 

This is how you generate the commit locally. For review, we prefer github these days, please see https://fedorahosted.org/sssd/wiki/GithubWorkflow

> Bugzilla and Fedora:
>     Since RHEL is fork of fedora, so do we need to own Bugzilla ticket also while owning the Fedora/sssd ticket.
>     Or We only change fedora/sssd ticket, code and it waterfalls to RHEL on upcoming release(if priotized)
> 

We can handle bugzilla, feel free to just submit the PR

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-16 10:15:07](https://pagure.io/SSSD/sssd/issue/2201#230552)**

Fields changed

owner: somebody => amitkumar25nov

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-12-01 07:57:16](https://pagure.io/SSSD/sssd/issue/2201#230553)**

Generated a pull request ..
https://github.com/SSSD/sssd/pull/98

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-01-06 11:51:13](https://pagure.io/SSSD/sssd/issue/2201#230554)**

master:

- dfbbe39f49ba3f7ff64370fecef971f091214d47

milestone: SSSD Patches welcome => SSSD 1.15 Alpha
resolution:  => fixed
status: new => closed

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 14:42:11](https://pagure.io/SSSD/sssd/issue/2201#230555)**

**Metadata Update from @jhrozek**:
- Issue assigned to amitkumar25nov
- Issue set to the milestone: SSSD 1.15.0
